core/math/transform_2d.cpp

```diff
@@ -173,10 +173,9 @@ bool Transform2D::is_finite() const {
 }
 
 Transform2D Transform2D::looking_at(const Vector2 &p_target) const {
-	Transform2D return_trans = Transform2D(get_rotation(), get_origin());
-	Vector2 target_position = affine_inverse().xform(p_target);
-	return_trans.set_rotation(return_trans.get_rotation() + (target_position * get_scale()).angle());
-	return return_trans;
+	const Vector2 origin = get_origin();
+	real_t target_angle = (p_target - get_origin()).angle();
+	return Transform2D(target_angle, origin);
```

Comment:
I'd say `looking_at` should rotate both basis vectors by the same angle just so `x` will point at the target (without changing their magnitude and angle between them). :thinking: E.g. like so (probably could be optimized):
```suggestion
	const Vector2 origin = get_origin();
	real_t angle = columns[0].angle_to(p_target - origin);
	return Transform2D(columns[0].rotated(angle), columns[1].rotated(angle), origin);
```